Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/FreerunningTest.java

```diff
@@ -0,0 +1,71 @@
+package org.mage.test.cards.abilities.keywords;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestCommanderDuelBase;
+
+/**
+ * @author TheElk801
+ */
+public class FreerunningTest extends CardTestCommanderDuelBase {
+
+    private static final String vision = "Eagle Vision";
+
+    @Test
+    public void testRegular() {
+        addCard(Zone.BATTLEFIELD, playerA, "Island", 5);
+        addCard(Zone.HAND, playerA, vision);
+
+        castSpell(1, PhaseStep.POSTCOMBAT_MAIN, playerA, vision);
+
+        setStrictChooseMode(true);
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertHandCount(playerA, 3);
+        assertTapped("Island", true);
```

Comment:
more robust to use assertTappedCount and check that all five Islands are tapped

also worth dealing combat damage with a non-Assassin non-commander

Reply:
good call